src/lib/common/types.ts

```diff
@@ -0,0 +1,15 @@
+import type { APIInteraction, APIPingInteraction } from 'discord-api-types/v10';
+
+export type AnyInteraction = Exclude<APIInteraction, APIPingInteraction>;
+export type IntegerString = `${bigint}`;
+
+// eslint-disable-next-line @typescript-eslint/ban-types
+export type NonNullObject = {} & object;
+
+export function isUndefined(value: unknown): value is undefined {
+	return value === undefined;
+}
+
+export function isDefined<T>(value: T | undefined): value is T {
+	return value !== undefined;
+}
```

Comment:
Add to deps and use @sapphire/utilities, it's already in node_modules through hfx -> sapphire pieces anyway so it's not like it'll add overhead.